src/debug/clean-pending-payments-for-dead-node.ts

```diff
@@ -0,0 +1,85 @@
+/**
+ * how to run:
+ *	. ./.envrc && yarn ts-node \
+ *		--files \
+ *			-r tsconfig-paths/register \
+ *			-r src/services/tracing.ts \
+ *		src/debug/clean-pending-payments-for-dead-node.ts
+ */
+
+import { LedgerTransactionType, UnknownLedgerError } from "@domain/ledger"
+
+import { isUp } from "@services/lnd/health"
+import { params as unauthParams } from "@services/lnd/unauth"
+
+import { setupMongoConnection } from "@services/mongodb"
+import { LedgerService, translateToLedgerTx } from "@services/ledger"
+import { LockService } from "@services/lock"
+import { baseLogger } from "@services/logger"
+
+import { MainBook } from "@services/ledger/books"
+
+const DEAD_LND_PUBKEYS = [
+  "02ae3c066e67cf3951bb5230cf4e13aee053bc06465cd25f870988b691469140b0", // dead node in hack environment
+]
+
+const listAllPendingPayments = async (): Promise<
+  LedgerTransaction<WalletCurrency>[] | LedgerError
+> => {
+  try {
+    const { results } = await MainBook.ledger({
+      type: LedgerTransactionType.Payment,
+      pending: true,
+    })
+
+    return results.filter((tx) => tx.debit > 0).map((tx) => translateToLedgerTx(tx))
+  } catch (err) {
+    return new UnknownLedgerError(err)
+  }
+}
+
+const main = async (): Promise<true | ApplicationError> => {
+  const pendingPayments = await listAllPendingPayments()
+  if (pendingPayments instanceof Error) return pendingPayments
+
+  for (const payment of pendingPayments) {
+    const { paymentHash, pubkey } = payment
+    if (pubkey === undefined) continue
+    if (paymentHash === undefined) continue
+
+    if (DEAD_LND_PUBKEYS.includes(pubkey)) {
+      await LockService().lockPaymentHash(
+        paymentHash,
+        async (): Promise<true | LedgerServiceError> => {
+          const ledgerService = LedgerService()
+          const settled = await ledgerService.settlePendingLnPayment(paymentHash)
+          if (settled instanceof Error) {
+            baseLogger.error({ error: settled }, "no transaction to update")
+            return settled
```

Comment:
we may want to `continue` here instead instead of `return`?

maybe we should also delete the record from the ledgerService?

Reply:
The `return`s inside here are for the `lockPaymentHash` internal function. They wouldn't exit the outer `for` loop I think (based on my testing so far locally).

Reply:
ohh yeah ok didn't pay attention to the nest anonymous function!